### ISSUE TITLE
XMP file names - possible to control the format?

### ISSUE THREAD

**Issue by narensankar0529** [RhetTbull/osxphotos#291]
Currently the way XMP files are named are after the extension -

For example -

ns_Canon_20170605_0017_2017_Jun_06.MOV
ns_Canon_20170605_0017_2017_Jun_06.MOV.xmp

But it appears that some software expects the naming as follows -

ns_Canon_20170605_0017_2017_Jun_06.MOV
ns_Canon_20170605_0017_2017_Jun_06.xmp

This means that currently the exported XMP are not read by those software and in some scenarios actually deleted. Mylio is one that does this, as well as some of Canon's tools

Is it possible to control the way the sidecars are named to support the second version where the file names have to be exactly the same with just the extension different?

**Comment by RhetTbull** [comment]
It's currently not possible to change the sidecar naming format but this could be added relatively easily with a command line option.  Originally, what you propose is how sidecar files were named. However, some users had issues (#210) with this because 1) some software like darktable expects the sidecar to have the photo extension.xmp and 2) this caused collisions where two files with different extensions but the same name were in a directory (for example, live images have both a .heic and .mov).  

I'll add a flag of some kind that tells osxphotos to drop the photo extension from sidecars if desired.

Perhaps `--sidecar-no-photo-ext`?

**Comment by narensankar0529** [comment]
Yes I figured that for live pictures this might be a problem. But seems like different software use different heuristics which creates issues. As I mentioned Mylio deletes all of the XMP because it thinks they are invalid and not pointing to valid pictures. Surprised that there isn't more of a standard here so supporting a different way to output the file would be ideal. This is especially true because osxphotos does a better job of extracting the dates for videos compared to other software looking at the mov files directly. I noticed that the xmp it produces is more accurate for the recorded date compared to even exiftool and I am guessing this is due to your ability to look at the DB directly. So having osxphotos export XMP is critical.

--sidecar-no-photo-ext sounds fine for this approach.
